tests/src/Interop/ICustomMarshaler/ConflictingNames/CustomMarshaler.cs

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+class WrappedString
+{
+    public WrappedString(string str)
+    {
+        _str = str;
+    }
+
+    internal string _str;
+}
+
+class WrappedStringCustomMarshaler : ICustomMarshaler
+{
+    public void CleanUpManagedData(object ManagedObj) { }
+    public void CleanUpNativeData(IntPtr pNativeData) { Marshal.ZeroFreeCoTaskMemAnsi(pNativeData); }
+
+    public int GetNativeDataSize() => IntPtr.Size;
+
+    public IntPtr MarshalManagedToNative(object ManagedObj) => Marshal.StringToCoTaskMemAnsi(((WrappedString)ManagedObj)._str);
+    public object MarshalNativeToManaged(IntPtr pNativeData) => new WrappedString(Marshal.PtrToStringAnsi(pNativeData));
+
+    public static ICustomMarshaler GetInstance(string cookie) => new WrappedStringCustomMarshaler();
+}
+
+#if CUSTOMMARSHALERS2
```

Comment:
Is this for a Framework vs Core split?

Reply:
Oh I see. This is to have distinct marshlers. That sounds good. Lets document what we are doing here. I am fine with the name, but if we want to extend/add more tests I would prefer it to be more obvious how to expand and the design so it can be done easily.